_scripts/customJS/tokenParser.js

```diff
@@ -0,0 +1,575 @@
+            if (obj2) for (var attrname in obj2) { obj3[attrname] = obj2[attrname]; }
+            return obj3;
+        }
+
+        // overrides that are used elsewhere //
+        // overrides.dateInfo replaces the dateInfo object that is computed by DateManager //
+
+        let metadata = merge_options(file.frontmatter, overrides)
+
+        if (targetDate) targetDate = DateManager.normalizeDate(targetDate)
+        let pageDateInfo = metadata.dateInfo ?? DateManager.getPageDates(metadata, targetDate)
+        let displayDefaults = NameManager.getDisplayData(metadata)
+        let sourcePageType = NameManager.getPageType(metadata)
+
+        let value = "";
+
+
+        // set formatting options //  
+        // define four types of formatters //
+        // none: no formatting, just return the value //
+        // name: apply name formatting, expects a nameObject object //
+        // date: apply date formatting, expects a normalizeDate object //
+        // whereabout: build a location chain from this starting point //
+        // casing: just apply casing //
+
+        let formatter = "none"
+
+        switch (token.token) {
+            // date options - value is expected to be a normalizedDate object //
+            // to do: actually check this //
+            case "startdate":
+                if (pageDateInfo.startDate) value = pageDateInfo.startDate
+                formatter = "date"
+                break;
+            case "targetdate":
+                value = targetDate ? targetDate : DateManager.getTargetDateForPage(metadata)
+                formatter = "date"
+                break;
+            case "enddate":
+                if (pageDateInfo.endDate) value = pageDateInfo.endDate
+                formatter = "date"
+                break;
+            case "lastknowndate":
+                let wb = WhereaboutsManager.getWhereabouts(metadata, targetDate)
+                if (wb.lastKnown.location && wb.lastKnown.awayEnd.sort <= targetDate.sort) {
+                    value = wb.lastKnown.awayEnd
+                } else {
+                    if (targetDate) {
+                        value = targetDate
+                    } else {
+                        value = DateManager.getTargetDateForPage(metadata)
+                    }
+                }
+                formatter = "date"
+                break;
+            // end date options //
+
+            // start name options - can be a string or a name object //
+            case "name":
+                value = metadata.name ? metadata.name : file.name
+                formatter = "name"
+                break;
+            case "ancestry":
+                value = metadata.ancestry
+                formatter = "name"
+                break
+            case "subtypeof":
+                value = metadata.subTypeOf
+                formatter = "name"
+                break
+            case "typeof":
```

Comment:
This alias of main type to typeof means that a page with species and speciesAlias won’t work correctly for main type. I think you need to get the name object here so you can pass the right alias.

Reply:
Yeah, I tested this, it doesn't work.